src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -0,0 +1,68 @@
+package org.corfudb.runtime.view;
+
+import org.corfudb.infrastructure.TestLayoutBuilder;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.runtime.clients.TestRule;
+import org.junit.Test;
+
+/**
+ * Test to verify the Management Server functionality.
+ * <p>
+ * Created by zlokhandwala on 11/9/16.
+ */
+public class ManagementViewTest extends AbstractViewTest {
+
+    /**
+     * Boolean flag turned to true when the MANAGEMENT_FAILURE_DETECTED message
+     * is sent by the Management client to its server.
+     */
+    private static volatile boolean failureDetectorMsgRecd;
+
+    /**
+     * Scenario with 3 nodes: 9000, 9001 and 9002.
+     * We fail 9000 and then listen to intercept the message
+     * sent by 9001's client to the server to handle the failure.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void invokeFailureHandler()
+            throws Exception {
+        addServer(9000);
+        addServer(9001);
+        addServer(9002);
+        Layout l = new TestLayoutBuilder()
+                .setEpoch(1L)
+                .addLayoutServer(9000)
+                .addLayoutServer(9001)
+                .addLayoutServer(9002)
+                .addSequencer(9000)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(9000)
+                .addLogUnit(9001)
+                .addLogUnit(9002)
+                .addToSegment()
+                .addToLayout()
+                .build();
+        bootstrapAllServers(l);
+
+        ManagementViewTest.failureDetectorMsgRecd = false;
+
+        // Adding a rule on 9000 to drop all packets
+        addServerRule(9000, new TestRule().always().drop());
+
+        // Adding a rule on 9001 to toggle the flag when it sends the
+        // MANAGEMENT_FAILURE_DETECTED message.
+        addClientRule(getManagementServer(9001).getCorfuRuntime(), new TestRule().matches(corfuMsg -> {
+            if (corfuMsg.getMsgType().equals(CorfuMsgType.MANAGEMENT_FAILURE_DETECTED)) {
+                ManagementViewTest.failureDetectorMsgRecd = true;
+            }
+            return true;
+        }));
+
+        // Wait in loop until failures have been reported.
+        while (!ManagementViewTest.failureDetectorMsgRecd)
+            Thread.sleep(500);
```

Comment:
Can you add a timeout to this loop? I don't want the test to hang if it fails somehow. Probably use a semaphore with a timeout on the acquire.